drivers/wireless/bluetooth/bt_rpmsghci_server.c

```diff
@@ -0,0 +1,547 @@
+ * Name: rpmsghci_ns_match
+ *
+ * Description:
+ *   Check if we match to the rpmsg name service.
+ *
+ ****************************************************************************/
+
+static bool rpmsghci_ns_match(FAR struct rpmsg_device *rdev, FAR void *priv,
+                              FAR const char *name, uint32_t dest)
+{
+  return !strncmp(name, RPMSGHCI_NAME_PREFIX, RPMSGHCI_NAME_PREFIX_LEN);
+}
+
+/****************************************************************************
+ * Name: rpmsghci_ept_cb
+ *
+ * Description:
+ *   Rpmsg HCI end point callback function, this function will be called
+ *   when receive the client cpu message.
+ *
+ * Parameters:
+ *   ept  - The rpmsg-HCI end point
+ *   data - The received data
+ *   len  - The received data length
+ *   src  - unknow
+ *   priv - unknow
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int rpmsghci_ept_cb(FAR struct rpmsg_endpoint *ept, FAR void *data,
+                           size_t len, uint32_t src, FAR void *priv)
+{
+  FAR struct rpmsghci_header_s *header  = data;
+  uint32_t                      command = header->command;
+
+  if (command < nitems(g_rpmsghci_handler))
+    {
+      return g_rpmsghci_handler[command](ept, data, len, src, priv);
+    }
+
+  return -EINVAL;
+}
+
+/****************************************************************************
+ * Name: rpmsghci_ns_unbind
+ *
+ * Description:
+ *   Unbind from the rpmsg name service.
+ *
+ ****************************************************************************/
+
+static void rpmsghci_ns_unbind(FAR struct rpmsg_endpoint *ept)
+{
+  rpmsg_destroy_ept(ept);
+
+  kmm_free(ept);
+}
+
+/****************************************************************************
+ * Name: rpmsghci_ns_bind
+ *
+ * Description:
+ *   Bind to the rpmsg name service.
+ *
+ ****************************************************************************/
+
+static void rpmsghci_ns_bind(FAR struct rpmsg_device *rdev, FAR void *priv,
+                             FAR const char *name, uint32_t dest)
+{
+  FAR struct rpmsghci_server_s *server = priv;
+
+  server->ept.priv = priv;
+  rpmsg_create_ept(&server->ept, rdev, name,
+                   RPMSG_ADDR_ANY, dest,
+                   rpmsghci_ept_cb, rpmsghci_ns_unbind);
+}
+
+/****************************************************************************
+ * Name: rpmsghci_bt_receive
+ *
+ * Description:
+ *   Called by the Bluetooth low-level driver when new data is received from
+ *   the radio.  Transfers data to the RPMSG client.
+ *
+ *   This callback should not be called from the interrupt context !
+ *
+ * Returned Value:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+int rpmsghci_bt_receive(FAR struct bt_driver_s *btdev,
```

Comment:
```suggestion
int rpmsghci_receive(FAR struct bt_driver_s *btdev,
```

Reply:
Due to the fact it is a replacement for `wireless/bluetooth/bt_hcicore.c::bt_receive()`, it's better to keep format `xxx_bt_receive()`